waifu2x.ts

```diff
@@ -295,34 +315,38 @@ export default class Waifu2x {
                 fileMap = [...fileMap, ...search]
             }
         }
-        if (!options.limit) options.limit = fileMap.length
-        const retArray: string[] = []
+
+        options.limit = options.limit ?? fileMap.length
         let cancel = false
         let counter = 1
         let total = fileMap.length
-        let queue: string[][] = []
-        if (!options.parallelFrames) options.parallelFrames = 1
-        while (fileMap.length) queue.push(fileMap.splice(0, options.parallelFrames))
         if (progress) progress(0, total)
-        for (let i = 0; i < queue.length; i++) {
-            await Promise.all(queue[i].map(async (f) => {
-                if (counter >= options.limit) cancel = true
-                const ret = await Waifu2x.upscaleImage(f, destFolder, options)
-                retArray.push(ret)
-                const stop = progress ? progress(counter++, total) : false
-                if (stop) cancel = true
-            }))
-            if (cancel) break
-        }
-        return retArray
+        const sem = new AsyncSemaphore(options.parallelFrames ?? 1)
+        const promises = fileMap.map(async (f) => {
+            return sem.add(async () => {
+                if(cancel) return null
+                if(counter >= options.limit) cancel = true
+                try{
+                    const ret = await Waifu2x.upscaleImage(f, destFolder, options)
+                    const stop = progress ? progress(counter++, total) : false
+                    if (stop) cancel = true
+                    return ret
+                }catch(e){

```

Comment:
Coding style:
-Put spaces between brackets and parentheses
-Label the error as `error` or `err`, not `e`.